sim/midas/src/main/scala/midas/passes/fame/LabelMultiThreadedInstances.scala

```diff
@@ -0,0 +1,31 @@
+// See LICENSE for license details.
+
+package midas.passes.fame
+
+import firrtl._
+
+// This pass labels all instances that are annotated as potential targets for threading as FAME models
+object LabelMultiThreadedInstances extends Transform {
+  def inputForm = HighForm
+  def outputForm = HighForm
+
+  override def execute(state: CircuitState): CircuitState = {
+    val p = state.annotations.collectFirst({ case midas.stage.phases.ConfigParametersAnnotation(p)  => p }).get
+    val enableMultiThreading = p(midas.EnableModelMultiThreading)
+    val fameModelAnnos = new collection.mutable.LinkedHashSet[midas.targetutils.FirrtlFAMEModelAnnotation]
+    val updatedAnnos = state.annotations.flatMap {
+      case fma: midas.targetutils.FirrtlFAMEModelAnnotation =>
+        fameModelAnnos += fma
+        Some(fma)
```

Comment:
Does this not double the occurrence of this annotation or am i mistaken? Is it not safe to make this None?

Reply:
Yes, fixed